rcl_interfaces/msg/IntegerRange.msg

```diff
@@ -0,0 +1,15 @@
+# Represents bounds and a step value for an integer typed parameter.
+
+# Lower bound for valid values, inclusive.
+int64 minimum_value
+
+# Upper bound for valid values, inclusive.
+int64 maximum_value
+
+# Size of valid steps between the lower and upper bound.
+# Step should be positive, and a step of zero implies a step size of one, i.e.
+# all integer values between the bounds are valid.
+# Step size should be less than the distance between the minimum and maximum
+# values, and ideally would be an even multiple of that distance, but that
+# is not required.
+int64 step
```

Comment:
Should this be unsigned?

Reply:
What's the reason above? What is the meaning of a negative step value?

Reply:
If we use `from / to` and allow invertible sliders, should we expect the step be positive or negative when `from > to`?

Reply:
How about having min and max. But using the sign in the step to represent direction?

Reply:
> How about having min and max. But using the sign in the step to represent direction?

That sound like a good idea to me. :+1:

Reply:
> If we use from / to and allow invertible sliders, should we expect the step be positive or negative when from > to?

The idea was to have step be a magnitude and therefore step would always be positive (unsigned). The relationship between from and to doesn't matter in that case.

> How about having min and max. But using the sign in the step to represent direction?

Personally I find that confusing, but I guess that's fine. You still need to check for min > max and do something in that case.

The nice thing about an unsigned step and from/to is that all of these instances are implicitly correct (`{from, to, step}`):

- `{0, 0, 0}`
- `{-1, 1, 0}`
- `{1, -1, 0}`
- `{-1, 1, 1}`
- `{1, 1, 1}`

And so on...

If you allow step to be > the distance between from and to, then there are no extra checks needed to consider it valid or not.

In the case of floating point you do have to additionally assert step is positive, unless you implicitly take the `fabs` of it always, in which case all values are possible again. You do have to have special logic to handle the direction of the slider, basically determining min/max based on from/to right before you render the GUI.

With min/max you need to additionally check that min < max and do different logic based on the sign of step instead of the ordering of from/to.

Reply:
I agree. I'd got with from / to and step >= 0, 0 meaning no step.